src/main/kotlin/lotto/core/WinningLogic.kt

```diff
@@ -0,0 +1,29 @@
+package lotto.core
+
+class WinningLogic(private val tickets: List<Lotto>, private val winningLotto: Lotto, private val bonusNumber: Int) {
+    //    fun matchLottos(ticket: lotto.core.Lotto, winningLotto: lotto.core.Lotto): Int {
+//        val matchingList = ticket.countMatches()
+//    }
+
+    fun determineWinningTickets(): Map<Rank, Int> {
+//        val mapOfResult = mutableMapOf<lotto.core.Rank, Int>()
+
+        return tickets
+            .map {
+                val countMatches = it.countMatches(winningLotto)
+                val hasBonus = it.containsBonus(bonusNumber)
+                Rank.valueOf(countMatches, hasBonus)
+            }.groupingBy { it }
+            .eachCount()
+
+//        tickets.forEach {
+//            val countMatches = it.countMatches(winningLotto)
+//            val hasBonus = it.containsBonus(bonusNumber)
+//            val rank = lotto.core.Rank.valueOf(countMatches, hasBonus)
+//            println(rank)
+//            mapOfResult.put(rank, mapOfResult.getOrDefault(rank, 0) + 1)
+//        }
+
+//        return mapOfResult
+    }
```

Comment:
Plz remove all unused comments.
Fellow developers might read the comments and wonder 'Should I refer to this?' which could cause confusion and reduce productivity. 😢